modules/metastorage/src/test/java/org/apache/ignite/internal/metastorage/server/BasicOperationsKeyValueStorageTest.java

```diff
@@ -2195,6 +2199,48 @@ void testClear() {
         assertThrows(CompactedException.class, () -> storage.revisionByTimestamp(hybridTimestamp(2)));
     }
 
+    @ParameterizedTest
+    @EnumSource(UpdateRevisionOperation.class)
+    void testNotifyUpdateRevisionForOperationAfterStartWatches(UpdateRevisionOperation updateRevisionOperation) {
+        var revisionUpdateListener = new TestRevisionUpdateListener();
+        storage.registerRevisionUpdateListener(revisionUpdateListener);
+
+        var watchEventHandlingCallback = new TestWatchEventHandlingCallback();
+        storage.startWatches(1, watchEventHandlingCallback);
+
+        storage.put(key(0), keyValue(0, 1), kvContext(hybridTimestamp(10)));
+
+        long revision = storage.revision();
+        long newRevision = revision + 1;
+
+        updateRevisionOperation.execute(storage);
+
+        assertThat(storage.revision(), equalTo(newRevision));
+        assertThat(revisionUpdateListener.get(newRevision), willSucceedFast());
```

Comment:
`willSucceedFast()` seems to be dangerous to use as an unfortunate GC pause might fail a test. `willCompleteSuccessfully()` looks safer